src/grid.c

```diff
@@ -3423,7 +3440,14 @@ SEXP L_points(SEXP x, SEXP y, SEXP pch, SEXP size)
     initGContext(currentgp, &gc, dd, gpIsScalar, &gcCache);
     nx = unitLength(x); 
     npch = LENGTH(pch);
-    nss = unitLength(size);
+    /*
+     * Need to take vector gpar elements into account that may affect unit size 
+     * calculations
+     */
+    nss = unitLength(size) * LENGTH(VECTOR_ELT(currentgp, GP_FONTSIZE)) * 
+        LENGTH(VECTOR_ELT(currentgp, GP_CEX)) * 
+        LENGTH(VECTOR_ELT(currentgp, GP_LINEHEIGHT));
+    nss = nss > nx ? nx : nss;
```

Comment:
Very simply `grid.points(default.units = 'npc', gp = gpar(fontsize = runif(10, min = 10, max = 20)))` should result in 10 different sizes of points because the single `1char` unit supplied for size is interpreted differently for each point due to the different font size. The old code didn't respect that as it only looked at the length of the size vector. In order to get correct recycling behaviour I'm multiplying the lengths of all relevant parameters (and capping it to the number of points)

Reply:
Cool.  And text does not need a similar change because a) it does not have a separate 'size' parameter and b) it is already recycling across font-related parameters (?)

Reply:
Yes exactly - sorry for not addressing that part of the question

Reply:
No problem.  Thanks for clarifying.  I'll try out the patch.

Reply:
You are regression testing this on 'ggplot2' right ? (my tests cover 'grid' and 'lattice' at the moment)

Reply:
Yes. No regressions (apart from the one I fixed herein)

Reply:
Great.  Thanks